src/chess/piece/Queen.java

```diff
@@ -14,11 +14,14 @@ public Queen(Color color) {
   }
 
   @Override
-  public boolean isValidMove(Position newPosition) {
-    if (!super.isValidMove(newPosition)) {
+  public boolean isValidMove(int row, int col, boolean isEnemyPiece) {
+    if (!super.isValidMove(row, col, isEnemyPiece)) {
       return false;
     }
-    // ToDo: implement
+
+    if (row == 0 && col == 0) return false;
```

Comment:
Maybe unnecessary as it should be blocked by `Board`
Or if you check this, this is not something only related to `Queen` so you should put this to the parent class